api/src/main/java/net/md_5/bungee/api/event/ServerChatEvent.java

```diff
@@ -0,0 +1,57 @@
+package net.md_5.bungee.api.event;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import net.md_5.bungee.api.ChatMessageType;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.connection.Connection;
+import net.md_5.bungee.api.plugin.Cancellable;
+import net.md_5.bungee.chat.ComponentSerializer;
+
+import java.util.function.Supplier;
+
+import static lombok.AccessLevel.NONE;
+
+/**
+ * Event called when a server sends a message to a player.
+ */
+@Data
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+public class ServerChatEvent extends TargetedEvent implements Cancellable
+{
+
+    /**
+     * Cancelled state.
+     */
+    private boolean cancelled;
+    /**
+     * Text contained in this chat.
+     */
+    private BaseComponent[] message = null;
+    @Getter(NONE)
+    @Setter(NONE)
+    private Supplier<BaseComponent[]> messageSupplier;
+    /**
+     * Type of message/display location
+     */
+    private ChatMessageType position;
+
+    public ServerChatEvent(Connection sender, Connection receiver, String json, byte position)
+    {
+        super( sender, receiver );
+        this.messageSupplier = () -> ComponentSerializer.parse( json );
+        this.position = ChatMessageType.values()[position];
+    }
+
+    public BaseComponent[] getMessage() {
```

Comment:
Racy. Maybe just chuck a synchronized on.

Reply:
Good shout, I didn't think about that